ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -305,24 +318,58 @@ func resourceIBMCbrRuleCreate(context context.Context, d *schema.ResourceData, m
 	return resourceIBMCbrRuleRead(context, d, meta)
 }
 
+// waitForCbrRuleRead will leverage use retry.StateChangeConf due to the service's eventual consistency
+func waitForCbrRuleRead(cbrClient *contextbasedrestrictionsv1.ContextBasedRestrictionsV1, context context.Context, id string, timeout time.Duration) (interface{}, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{cbrRuleReadPending},
+		Target:  []string{cbrRuleReadError, cbrRuleReadComplete, ""},
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[DEBUG] Read cbr rule (%s) read", id)
+			getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
+			getRuleOptions.SetRuleID(id)
+			_, response, err := cbrClient.GetRuleWithContext(context, getRuleOptions)
+			if err != nil && response.StatusCode == 404 {
+				return response, cbrRuleReadPending, nil
+			} else if err != nil {
+				return response, cbrRuleReadError, err
+			} else {
+				return response, cbrRuleReadComplete, nil
+			}
+		},
+		Timeout:    timeout,
+		Delay:      20 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+	return stateConf.WaitForStateContext(context)
+}
+
 func resourceIBMCbrRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
-
 	getRuleOptions.SetRuleID(d.Id())
 
 	rule, response, err := contextBasedRestrictionsClient.GetRuleWithContext(context, getRuleOptions)
 	if err != nil {
+		//  Manual change. leverage retry for the read due to eventual consistency of the service.
 		if response != nil && response.StatusCode == 404 {
-			d.SetId("")
```

Comment:
Chane the retry logic to use Resource.RetryContext 
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/b8c55bb204c494ff3bf049521e75c92c473f5dd4/ibm/service/iamaccessgroup/resource_ibm_iam_access_group.go#L90

Reply:
Resource.RetryContext is considered deprecated.
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#RetryableError